src/Button.tsx

```diff
@@ -0,0 +1,291 @@
+  disabledBackgroundColor = colors.silver.dark,
+  disabledTextColor = colors.grey.light,
+  fab = false,
+  forceActiveState = false,
+  forceFocusState = false,
+  forceHoverState = false,
+  hidden = false,
+  hiddenActiveBackgroundColor = colors.silver.dark,
+  hiddenFocusedColor = colors.blue.base,
+  hoverBackgroundColor = colors.silver.base,
+  icon,
+  size = "default",
+  ...otherProps
+}) => {
+  /**
+   * Icon size in pixels
+   *
+   * This is stored so we can use the same value for `height` and `width`
+   */
+  const iconSize = size === "small" ? 12 : size === "large" ? 24 : 16;
+
+  const iconOnly = !children;
+
+  if (fab) {
+    if (!icon) {
+      throw new TypeError("FAB buttons are required to have an `icon`");
+    } else if (children) {
+      throw new TypeError("FAB buttons cannot have children, only an `icon`");
+    }
+  }
+
+  if (forceActiveState && forceHoverState) {
+    throw new TypeError(
+      "Do not force multiple properties at once, you will get indeterminiate behavior"
+    );
+  }
+
+  // We're storing this because it can be used for the `:active` style or when
+  // `forceActiveState` is used in Storybook
+  const activeProperties = {
+    ...(hidden
+      ? { backgroundColor: hiddenActiveBackgroundColor }
+      : {
+          // The `box-shadow` property is copied directly from Zeplin
+          boxShadow:
+            "inset 0 0 0 1px rgba(18, 21, 26, 0.2), inset 0 -1px 0 0 rgba(18, 21, 26, 0.05), inset 0 2px 2px 0 rgba(18, 21, 26, 0.12)"
+        }),
+    outline: 0
+  };
+
+  // We're storing this because it can be used for the `:hover` style or when
+  // `forceHoverState` is used in Storybook
+  const hoverProperties = {
+    backgroundColor: hoverBackgroundColor,
+    cursor: "pointer",
+    ...(!hidden && {
+      // The `box-shadow` property is copied directly from Zeplin
+      boxShadow:
+        "0 5px 10px 0 rgba(18, 21, 26, 0.12), inset 0 0 0 1px rgba(18, 21, 26, 0.2), inset 0 -1px 0 0 rgba(18, 21, 26, 0.05)"
+    })
+  };
+
+  // We're storing this because it can be used for the `:focus` style or when
+  // `forceFocusState` is used in Storybook
+  const focusProperties = {
+    // The `box-shadow` property is copied directly from Zeplin
+    boxShadow:
+      "0 1px 4px 0 rgba(18, 21, 26, 0.08), 0 0 0 2px #bbdbff, inset 0 0 0 1px #2075d6, inset 0 -1px 0 0 rgba(18, 21, 26, 0.05)",
+    ...(hidden && { color: `${hiddenFocusedColor} !important` })
+  };
+
+  return (
+    <button
+      {...otherProps}
+      css={{
+        appearance: "button",
+        backgroundColor: hidden
+          ? "transparent"
+          : disabled
+          ? disabledBackgroundColor
+          : backgroundColor,
+        borderRadius: fab ? "100%" : 4,
+        borderWidth: 0,
+        ...(!hidden && {
+          boxShadow:
+            "0 1px 4px 0 rgba(18, 21, 26, 0.08), inset 0 0 0 1px rgba(18, 21, 26, 0.2), inset 0 -1px 0 0 rgba(18, 21, 26, 0.05)"
+        }),
+        minWidth: iconOnly
```

Comment:
Instead of using these ternaries within our object definition, maybe it would be more easily understandable if we break these logical statements out to their own functions. For example:

```js
function getMinWidth(iconOnly, size) {
  switch(size) {
    case 'small':
      return iconOnly ? 28 : 76;
    case 'large':
      return iconOnly ? 42 : 112;
    default:
      return iconOnly ? 36 : 100;
  }
}

<button
  css={{
    minWidth: getMinWidth(iconOnly, size)
  }}
/>
```

Reply:
I'm going to leave this as it is now because I think it boils down to personal preference and could end up in bikeshedding. If we find need to revisit this because we have problems, then let's revisit it.